tests/integration/components/file-browser/component-test.ts

```diff
@@ -245,4 +245,41 @@ module('Integration | Component | file-browser', hooks => {
             assert.dom(`[data-test-file-list-link][aria-label="${newFolderAria}"]`)
                 .exists('Shiny new folder exists');
         });
+
+    test('it creates new folder using enter key',
+        async function(this: FileBrowserTestContext, assert) {
+            const node = await this.store.findRecord('node', this.mirageNode.id);
+            const storageProviders = await node.files;
+            this.osfStorageProvider = storageProviders.toArray()[0];
+            await render(hbs`
+                <StorageProviderManager::StorageManager @provider={{this.osfStorageProvider}} as |manager|>
+                    <FileBrowser @manager={{manager}} @enableUpload={{true}} @selectable={{true}} />
+                </StorageProviderManager::StorageManager>
+            `);
+            await click('[data-test-add-new-trigger]');
+            await click('[data-test-create-folder]');
+            assert.dom('[data-test-create-folder-heading]').exists('Create folder modal shown');
+            assert.dom('[data-test-new-folder-error]').containsText(
+                t('osf-components.file-browser.create_folder.error_message'), 'Message shown to enter folder name',
+            );
+            assert.dom('[data-test-create-folder-button]').isDisabled('Create folder button is disabled');
+            await triggerKeyEvent('[data-test-create-folder-main] input', 'keyup', 'Enter');
+            assert.dom('[data-test-create-folder-heading]')
+                .exists('Create folder modal still visible after enter when folder name is empty');
+
+            await fillIn('[data-test-create-folder-main] input', '     ');
+            assert.dom('[data-test-new-folder-error]').containsText(
+                t('osf-components.file-browser.create_folder.error_message'), 'Folder cannot be empty',
+            );
+            await triggerKeyEvent('[data-test-create-folder-main] input', 'keyup', 'Enter');
+            assert.dom('[data-test-create-folder-heading]')
+                .exists('Create folder modal still visible after enter when folder name is just spaces');
+
+            await fillIn('[data-test-create-folder-main] input', 'Shiny New Folder');
+            assert.dom('[data-test-new-folder-error]').doesNotExist('No error message shown');
+            assert.dom('[data-test-create-folder-button]').isEnabled('Create folder button is enabled');
+            await triggerKeyEvent('[data-test-create-folder-main] input', 'keyup', 'Enter');
```

Comment:
So, for whatever reason, `keydown` or `keypress` here was not triggering the createFolder behavior. Not sure why...